datamodel/src/main/kotlin/com/hartwig/actin/datamodel/clinical/SequencingTest.kt

```diff
@@ -33,11 +33,13 @@ data class SequencedFusion(
     val exonDown: Int? = null
 ) : Displayable {
     override fun display(): String {
-        return when {
-            geneUp != null && geneDown == null -> "$geneUp fusion"
-            geneUp == null && geneDown != null -> "$geneDown fusion"
-            geneUp != null && geneDown != null -> "$geneUp::$geneDown fusion"
-            else -> throw IllegalStateException("Both genes in fusion are null")
+        return buildString {
+            if (geneUp != null) append("$geneUp")
+            if (exonUp != null) append(" exon $exonUp")
+            if (geneUp != null && geneDown != null) append(" :: ")
+            if (geneDown != null) append("$geneDown")
+            if (exonDown != null) append(" exon $exonDown")
+            append(" fusion")
         }
     }
```

Comment:
This seems to work, but I usually avoid string builders:
```suggestion
        return listOfNotNull(
            formatGeneAndExon(geneUp, exonUp), formatGeneAndExon(geneDown, exonDown)
        ).joinToString(" :: ") + " fusion"
    }
        
    fun formatGeneAndExon(gene: String, exon: String) = listOfNotNull(gene, exon?.let { "exon $it" }).joinToString(" ").ifEmpty { null }
```

Reply:
I like it, thanks! Since I don't use :: always I had to make a few changes, see PR :)